tedana/decay.py

```diff
@@ -97,6 +119,30 @@ def fit_decay(data, tes, mask, masksum):
         t2ss[..., echo - 1] = np.squeeze(utils.unmask(t2s, mask))
         s0vs[..., echo - 1] = np.squeeze(utils.unmask(s0, mask))
 
+    if fittype == 'curvefit':
+        # perform a monoexponential fit of echo times against MR signal
+        # using the mean of the signal, using loglin estimate
+        # as initial starting points for fit
+        
+        t2scf = np.zeros([t2s.size,])
+        s0cf = np.zeros([t2s.size,])
+
+        for voxel in range(0, t2s.size):
+            try:
+                popt, cov = scipy.optimize.curve_fit(mono_exp, tes, fit_data[voxel, :],
+                                                     p0=(s0[voxel],
+                                                     t2s[voxel]))
+                s0cf[voxel] = popt[0]
+                t2scf[voxel] = popt[1]
+            except RuntimeError:
```

Comment:
Could you add a comment explaining what might trigger the RuntimeError?

Reply:
Done - sometimes this fails to converge - and that will crash out - added a comment